heksher/db_logic/metadata.py

```diff
@@ -16,40 +16,42 @@
+                     Column('setting', ForeignKey(settings.columns.name,
+                                                  ondelete="CASCADE", onupdate="CASCADE"), index=True),
                      Column('context_feature', ForeignKey(context_features.columns.name)),
                      UniqueConstraint('setting', 'context_feature')
                      )
 
 rules = Table('rules', metadata,
               Column('id', Integer, primary_key=True, autoincrement=True),
-              Column('setting', ForeignKey(settings.columns.name, ondelete="CASCADE"), index=True),
+              Column('setting', ForeignKey(settings.columns.name, ondelete="CASCADE", onupdate="CASCADE"), index=True),
               Column('value', String, nullable=False),
               )
 
 conditions = Table('conditions', metadata,
-                   Column('rule', ForeignKey(rules.columns.id, ondelete="CASCADE"), index=True),
+                   Column('rule', ForeignKey(rules.columns.id, ondelete="CASCADE", onupdate="CASCADE"), index=True),
                    Column('context_feature', ForeignKey(context_features.columns.name)),
                    Column('feature_value', String, nullable=False),
                    UniqueConstraint('rule', 'context_feature'),
                    )
 
 setting_metadata = Table('setting_metadata', metadata,
-                         Column('setting', ForeignKey(settings.columns.name, ondelete="CASCADE"), index=True),
+                         Column('setting', ForeignKey(settings.columns.name,
+                                                      ondelete="CASCADE", onupdate="CASCADE"), index=True),
                          Column('key', String, nullable=False),
                          Column('value', JSONB, nullable=False),
                          UniqueConstraint('setting', 'key'),
                          )
 
 rule_metadata = Table('rule_metadata', metadata,
-                      Column('rule', ForeignKey(rules.columns.id, ondelete="CASCADE"), index=True),
+                      Column('rule', ForeignKey(rules.columns.id, ondelete="CASCADE", onupdate="CASCADE"), index=True),
                       Column('key', String, nullable=False),
                       Column('value', JSONB, nullable=False),
                       UniqueConstraint('rule', 'key'),
                       )
 
 setting_aliases = Table('setting_aliases', metadata,
-                        Column('setting', ForeignKey(settings.columns.name, ondelete="CASCADE"), index=True),
-                        Column('alias', String, nullable=False),
-                        UniqueConstraint('setting', 'alias'),
+                        Column('setting', ForeignKey(settings.columns.name,
+                                                     ondelete="CASCADE", onupdate="CASCADE"), index=True),
+                        Column('alias', String, nullable=False, unique=True, index=True),
```

Comment:
on second look, isn't this technically a primary key? https://www.postgresql.org/docs/14/ddl-constraints.html#DDL-CONSTRAINTS-PRIMARY-KEYS